api/proto/auth/client.go

```diff
@@ -0,0 +1,113 @@
+package auth
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sync"
+	"sync/atomic"
+
+	"github.com/gravitational/roundtrip"
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+var log = logrus.WithFields(logrus.Fields{
+	trace.Component: teleport.ComponentAPI,
+})
+
+// Client is a grpc Client that connects to a teleport auth server through TLS.
+type Client struct {
+	Cfg     Config
```

Comment:
It's not very customary in Go to shorthand to Cfg, I'd either name the field Config or just embed it.